Integrations/integration-ATD.yml

```diff
@@ -37,672 +38,670 @@ configuration:
-
-    function sendRequest(uri, method, headers, body){
-        body = sendRequestRaw(uri,method, headers, body).Body;
-        var res = JSON.parse(body);
-        if (res.success) {
-            if (res.success === 'false') {
-                throw 'ATD Api call to '+uri+' failed with request body: '+body;
+        return prettyCurrentUser
+
+    def get_session():
+        res = http_request('php/session.php','get', LOGINHEADERS)
+        md = tableToMarkdown('ATD Current User',prettify_current_user_res(res['results']),
+                                ['APIVersion', 'IsAdmin', 'SessionId', 'UserId'])
+        demisto.results({
+            'Type': entryTypes['note'],
+            'ContentsFormat': formats['json'],
+            'Contents': res['results'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': md,
+            'EntryContext': {
+                  "ATD.Session(val.SessionId==obj.SessionId)": prettify_current_user_res(res['results'])
             }
+        })
+
+    def prettify_list_users_res(users):
+        prettyUsers = []
+
+        for i in range(len(users)):
+            user = users[i]
+            prettyUsers.append({
+                'FullName': user['fullName'],
+                'UserId': user['idx'],
+                'LoginId': user['loginId'],
+                'UserType': user['userType']
+            })
+
+        return prettyUsers
+
+    def list_users(userType):
+        userType = userType if userType else 'STAND_ALONE'
+        result = http_request('php/briefUserList.php?userType='+userType,'get', API_HEADERS)
+        users = result['results']
+        return users
+
+    def list_users_command():
+        users = list_users(demisto.args()['userType'])
+
+        prettyUsers = prettify_list_users_res(users)
+        md = tableToMarkdown(
+            'ATD User List',
+            prettyUsers,
+            ['FullName', 'UserId', 'LoginId', 'UserType']
+            )
+
+        demisto.results({
+            'Type': entryTypes['note'],
+            'ContentsFormat': formats['json'],
+            'Contents': users,
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': md,
+            'EntryContext': {
+                  "ATD.Users(val.UserId==obj.UserId)": prettyUsers,
+            }
+        })
+
+    def prettify_list_profiles_res(profiles):
+        prettyProfiles = []
+        for i in range(len(profiles)):
+            prettyProfiles.append({
+                'Name': profiles[i]['name'],
+                'AnalyzerProfileId': profiles[i]['vmProfileid'],
+                'Description': profiles[i]['vmDesc'],
+                'Sandbox': 'True' if profiles[i]['sandbox'] == 1 else 'False',
+                'Internet': 'True' if profiles[i]['internet'] == 1 else 'False',
+                'LocalBlackList': 'True' if profiles[i]['locBlackList'] == 1 else 'False'
+            })
+        return prettyProfiles
+
+    def list_profiles():
```

Comment:
Please split to command and requests functions.

Reply:
done